subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/catalog/KotlinDslVersionCatalogExtensionIntegrationTest.groovy

```diff
@@ -375,6 +375,69 @@ class KotlinDslVersionCatalogExtensionIntegrationTest extends AbstractHttpDepend
         succeeds ':checkDeps'
     }
 
+
+    @Issue("https://github.com/gradle/gradle/issues/22650")
+    def "can use the generated extension to declare a dependency constraint with and without sub-group using bundles"() {
+        settingsKotlinFile << """
+            dependencyResolutionManagement {
+                versionCatalogs {
+                    register("libs") {
+                        version("myLib") {
+                            strictly("[1.0,1.1)")
+                        }
+                        library("myLib", "org.gradle.test", "lib-core").versionRef("myLib")
+                        library("myLib-ext", "org.gradle.test", "lib-ext").versionRef("myLib")
+                        bundle("myBundle", listOf("myLib"))
+                        bundle("myBundle-ext", listOf("myLib-ext"))
+                    }
+                }
+            }
+        """
+        def publishLib = { String artifactId, String version ->
+            def lib = mavenHttpRepo.module("org.gradle.test", artifactId, version)
+                .withModuleMetadata()
+                .publish()
+            lib.moduleMetadata.expectGet()
+            lib.pom.expectGet()
+            return lib
+        }
+        publishLib("lib-core", "1.0").with {
+            it.rootMetaData.expectGet()
+            it.artifact.expectGet()
+        }
+        publishLib("lib-core", "1.1")
+        publishLib("lib-ext", "1.0").with {
+            it.rootMetaData.expectGet()
+            it.artifact.expectGet()
+        }
+
+        withCheckDeps()
+        buildKotlinFile << """
+            plugins {
+                `java-library`
+            }
+
+            dependencies {
+                implementation("org.gradle.test:lib-core:1.+") // intentional!
+                implementation("org.gradle.test:lib-ext") // intentional!
+                constraints {
+                    implementation(libs.bundles.myBundle)
+                    implementation(libs.bundles.myBundle.ext)
+                }
+            }
+
+            tasks.register<CheckDeps>("checkDeps") {
+                files.from(configurations.compileClasspath)
+                expected.set(listOf("lib-core-1.0.jar", "lib-ext-1.0.jar"))
+            }
+            // Might be worth checking constraints too? Not sure if necessary because the Groovy DSL version covers that
+            // and the selected versions above would be wrong.
```

Comment:
💭 How would you even check constraints here, as this doesn't use the ResolveTestFixture like the Groovy DSL `VersionCatalogExtensionIntegrationTest`?

I think the javadoc on this class suggests against adding that check:

> This test isn't meant to check the behavior of the extension generation like the other integration tests in this package, but only what is very specific to the Kotlin DSL.

Reply:
My review is too late!  Probably still worth cleaning up this comment.